tools/wptrunner/wptrunner/testdriver-extra.js

```diff
@@ -43,10 +43,19 @@
         } else if (data.type === "testdriver-event") {
             const event_data = JSON.parse(data.message);
             const event_name = event_data.method;
-            const event = new Event(event_name);
-            event.payload = event_data.params;
-            event_target.dispatchEvent(event);
+            const testdriver_event = new Event(event_name);
+            testdriver_event.payload = event_data.params;
+            event_target.dispatchEvent(testdriver_event);
+        } else {
+            return;
         }
+
+        // Don't expose testdriver.js-internal messages to tests. Because
+        // `testdriver.js` precedes test scripts in the markup, this "message"
```

Comment:
Should the event handler be registered as a capturing handler  (third argument true) for extra assurance?

Reply:
I think it's not necessary. [MDN] says the `message` event doesn't bubble.

[MDN]: https://developer.mozilla.org/en-US/docs/Web/API/Window/message_event